### ISSUE TITLE
Link hint values change on page load

### ISSUE THREAD

**Issue by Delgan** [gdh1995/vimium-c#1074]
**Describe the bug**

Hi.

I'm facing a recurring annoying problem. I open a page, I quickly type "F" to display link hints, I prepare to follow the link, but then the hint changes.

I'm quite sure this is because of JavaScript updating something in the DOM. 

I don't know the algorithm used to generate the link hint values, but could we detect that the link didn't change and don't generate a new hint value?

**To Reproduce**

1. Go to https://github.com/gdh1995/vimium-c/issues/666
2. Type "F" to display link hints before the page is fully loaded
3. Observe that the values of link hints changes

https://github.com/gdh1995/vimium-c/assets/4193924/3778df2d-0a79-4455-a37a-031cf99cbafc

**Browser, OS and Vimium C versions**

- Browser name: Firefox
- Browser version: 121.0.1
- Vimium C version: 1.99.997
- OS name and version: Linux 6.6.10-arch1-1

**Comment by gdh1995** [comment]
Yes Vimium C detects "some cases" and re-inits `LinkHints` on page loading.

Up to now Vimium C can not make hint letters stable during page changes.

You may turn off this feature by `map f LinkHints.activate autoReload=false`

**Comment by Delgan** [comment]
Thanks for the explanation and possible workaround.

> Up to now Vimium C can not make hint letters stable during page changes.

Ok, I understand. Therefore, this ticket is more of a "feature request", but if it's not technically feasible or too complex, feel free to close this ticket. I can live with it, it's a minor inconvenience compared to the benefits of using Vimium. :)
